README.md

```diff
@@ -1,2 +1,132 @@
+    parser.add_argument('--threads', dest='threads', action='store',
+                        default=10, type=int, help="Number of threads to spawn")
+    default_command_line_args(parser)
+
+    args = parser.parse_args()
+
+    site = args.site
+    prefix = args.prefix
+    threads = args.threads
+    logger.info("ServiceX DID Finder starting up: "
+                f"Threads: {threads} Site: {site} Prefix: {prefix}")
+
+    # Initialize the finder
+    did_client = DIDClient()
+    replica_client = ReplicaClient()
+    rucio_adapter = RucioAdapter(did_client, replica_client)
+
+    # Run the DID Finder
+    try:
+        logger.info('Starting rucio DID finder')
+
+        async def callback(did_name, info):
+            async for f in find_files(rucio_adapter, site, prefix, threads,
+                                      did_name, info):
+                yield f
+
+        start_did_finder('rucio',
+                         callback,
+                         parsed_args=args)
+
+    finally:
+        logger.info('Done running rucio DID finder')
+```
+
+In particular note:
+
+1. The call to `default_command_line_args` to setup the arguments required by the finder library.
+2. Parsing of the arguments using the usual `parse_args` method
+3. Passing the parsed arguments to `start_did_finder`.
+
+Another pattern in the above code that one might find useful - a thread-safe way of passing global arguments into the callback. Given Python's Global Interpreter Lock, this is probably not necessary.
+
+### Proper Logging
+
+In the end, all DID finders for ServiceX will run under Kubernetes. ServiceX comes with a built in logging mechanism. If anything is to be logged it should use the log system using the python standard `logging` module, with some extra information. For example, here is how to log a message from your callback function:
+
+```python
+    import logger
+    __log = logger.getLogger(__name__)
+    async def my_callback(did_name: str, info: Dict[str, Any]):
+        __log.info(f'Looking up dataset {did_name}.',
+                     extra={'requestId': info['request-id']})
+
+        for i in range(0, 10):
+            yield {
+                'file_path': f"root://atlas-experiment.cern.ch/dataset1/file{i}.root",
+                'adler32': b183712731,
+                'file_size': 0,
+                'file_events': 0,
+            }
+```
+
+Note the parameter `request-id`: this marks the log messages with the request id that triggered this DID request. This will enable the system to track all log messages across all containers connected with this particular request id - making debugging a lot easier.
+
+The `start_did_finder` will configure the python root logger properly to dump messages with a request ID in them.
+
+*NOTE*: Once `start_did_finder` is called, whenever you log a message you _must_ pass the extra argument. If not, a formatting exception will occur during logging and the message will be swallowed.
```

Comment:
This part isn't true anymore if you're using the DID Formatter class.  That will automatically fill in None for the request id if it's not in the extra argument or if the extra argument is not given.

Reply:
Ah! I didn't realize how that worked. Will update!